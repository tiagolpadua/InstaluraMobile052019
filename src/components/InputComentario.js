import React, { Component } from 'react';
import { StyleSheet, View, Image, TouchableOpacity, TextInput } from 'react-native';

export default class InputComentario extends Component {
  constructor() {
    super();
    this.state = {
      valorComentario: '',
    };
  }

  render() {
    return (
      <View style={styles.novoComentario}>
        <TextInput
          style={styles.input}
          placeholder="Adicione um comentÃ¡rio..."
          ref={input => {
            this.inputComentario = input;
          }}
          onChangeText={texto => this.setState({ valorComentario: texto })}
          underlineColorAndroid="transparent"
        />

        <TouchableOpacity
          onPress={() => {
            const { valorComentario } = this.state;
            const { comentarioCallback } = this.props;
            comentarioCallback(valorComentario, this.inputComentario);
            this.setState({ valorComentario: '' });
          }}
        >
          <Image style={styles.icone} source={require('../../resources/img/send.png')} />
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  novoComentario: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderBottomColor: '#ddd',
  },
  input: {
    flex: 1,
    height: 40,
  },
  icone: {
    width: 30,
    height: 30,
  },
});
