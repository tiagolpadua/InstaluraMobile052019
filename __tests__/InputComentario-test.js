/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InputComentario from '../src/components/InputComentario';

it('Deve renderizar corretamente', () => {
  renderer.create(<InputComentario idFoto={1} comentarioCallback={() => {}} />);
});
