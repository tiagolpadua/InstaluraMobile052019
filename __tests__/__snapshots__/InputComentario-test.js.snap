// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar corretamente 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#ddd",
      "borderBottomWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Adicione um comentÃ¡rio..."
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../resources/img/send.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</View>
`;
